### DISCUSSION TITLE
How to cite inside a raw Latex table ?

### DISCUSSION THREAD

**Discussion by avhz** [quarto-dev/quarto-cli#6394] [Q&A] [answered]
### Description

I have a Latex table inside my Quarto document (PDF output), and I am trying to cite something from my bibliography, but just end up with the question marks indicating that Latex can't find my bibliography (I think). 

Citations elsewhere in my document work fine (I use the Quarto `[@citation]` format elsewhere), but using `\cite{}` does not seem to work within the Latex section (nor does the Quarto format).

The Latex table is like:

```qmd
\newcommand\ytl[2]{
    \parbox[b]{8em}{
        \hfill{\color{cyan}\bfseries\sffamily #1}~$\cdots\cdots$~
    }
    \makebox[0pt][c]{$\bullet$}\vrule\quad
    \parbox[c]{10cm}{
        \vspace{7pt}
        \color{red!40!black!80}
        \raggedright\sffamily #2 \\[7pt]
    }\\[-3pt]
}
\begin{table}[!h]
\caption{\label{timeline}Table with failing citations.}
\begin{minipage}[t]{1\linewidth}
\color{gray}
\rule{\linewidth}{1pt}
    \ytl{1995}{Attempting to cite: \cite{@citation}}            % <-- HERE
\rule{\linewidth}{1pt}
\end{minipage}
\end{table}
```

I generate my bibliography using the refs div:

```qmd
::: {#refs}
:::
```

I am not sure what else to include since citations work fine everywhere else.

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by avhz** [comment]
````qmd
---
title: "Citations Not Working"
format: pdf
bibliography: bib.bib
---

Here is a working citation: [@citation]

Here is not a working citation: \cite{citation}

\newcommand{\ytl}[2] {
    \parbox[b] { 8em } {
        \hfill{\color{gray}\bfseries\sffamily #1}~$\cdots\cdots$~
    }
    \makebox[0pt][c] { $\bullet$ }\vrule\quad
    \parbox[c] { 12cm } {
        \vspace{7pt} \color{black} \raggedright \sffamily #2.\\[7pt] 
    }\\[-3pt]
}
\begin{table}[!h]
\caption{\label{timeline}History of Citation Attempts.}
\begin{minipage}[t]{1\linewidth}
\color{gray}
\rule{\linewidth}{1pt}
\ytl { 2023 } {
    Attempt to cite \cite{citation}
}
\ytl { 2023 } {
    Another attempt to cite @citation
}
\rule{\linewidth}{1pt}
\end{minipage}
\end{table}

# References

\bibliography{bib}

::: {#refs}
:::
````

With `bib.bib`:

```
@article{citation,
    author = {Citer},
    year = 2023
}
```

Produces:


<img width="352" alt="Screenshot 2023-08-05 at 15 08 06" src="https://github.com/quarto-dev/quarto-cli/assets/63152017/1f4afed9-0e93-4cbc-a8a0-ecbdcac79f0c">

**Reply by avhz** [reply]
So basically, I cannot figure out how to use a raw LaTeX citation like `\cite{citation}` instead of Pandoc's `@citation` format. 

I need to use a raw LaTeX citation inside the table since the Pandoc format won't work inside the environment.

**Reply by baptiste** [reply]
For reference, [this answer has some interesting discussion](https://tex.stackexchange.com/questions/686867/reference-citation-in-latex-block-in-pandoc)

**Comment by avhz** [comment, marked as answer]
I changed to `cite-method: natbib` and used `\citet{}` and `\citep{}` instead. 

However I am still interested if there are other solutions if you know of any.

**Reply by mcanouil** [reply]
Yes, but it uses HTML table as the input, see <https://quarto.org/docs/authoring/tables.html#html-tables>.

**Reply by avhz** [reply]
What uses HTML table as input ?

**Reply by mcanouil** [reply]
Quarto can use HTML tables as described which internally is read in the AST format then translated to the right output.

**Comment by cscheid** [comment]
I wonder if Quarto could support a (fake) LaTeX command like

```latex
\QuartoMarkdown{We will convert this `Markdown` to LaTeX for you, including citations to @cite-foo and references like @fig-bar}
```

It would be hard to make this perfectly robust, but we could search for the text in `latex` RawBlocks and RawInline elements, and treat it similarly to how we treat `data-qmd` spans (see https://quarto.org/docs/authoring/tables.html#html-tables).

**Reply by mcanouil** [reply]
If it's build on top of `verbatim` this might work.

**Reply by cscheid** [reply]
The cool thing here is that the `\QuartoMarkdown` LaTeX command would never even be seen by LaTeX, because we would intercept it ahead of time. It's just a way for us to get a "LaTeX-looking" command that we catch and process in our Lua filters.

**Reply by cscheid** [reply]
This is exactly in the same way that a web browser never sees a `<span data-qmd=""></span>`; we catch it and transform it ourselves. `<span>...` is just a way to satisfy the HTML parser in Pandoc. It wouldn't work quite the same in LaTeX because we would never send that through the LaTeX reader, but from the user's perspective it's the same thing: a way to encapsulate quarto markdown in a different language.

**Reply by cscheid** [reply]
(Tracking here: https://github.com/quarto-dev/quarto-cli/issues/6503)
